src/main/java/org/ossgang/commons/observables/DispatchingObservableValue.java

```diff
@@ -51,18 +56,45 @@ public Subscription subscribe(Observer<? super T> listener, SubscriptionOption..
         Set<SubscriptionOption> optionSet = new HashSet<>(Arrays.asList(options));
         Subscription subscription = super.subscribe(listener, options);
         if (optionSet.contains(FIRST_UPDATE)) {
-            Optional.ofNullable(lastValue.get()).ifPresent(uncheckedConsumer(value -> dispatch(listener::onValue, value).get()));
+            Optional.ofNullable(lastValue.get())
+                    .ifPresent(uncheckedConsumer(value -> dispatch(listener::onValue, value).get()));
         }
         return subscription;
     }
 
     @Override
     protected void dispatchValue(T newValue) {
         requireNonNull(newValue, "null value not allowed");
```

Comment:
This check we can remove - accumulate does it.

Reply:
Changed in latest commit